features/FxA-83-sign-in-confirmation-email/README.md

```diff
@@ -0,0 +1,290 @@
+
+4. Auth server sends same `tokenVerificationId`
+   and `tokenVerified:false`
+   in request to `PUT /keyFetchToken/:tokenId`.
+
+5. Auth server sends response back to content server.
+
+6. Content server redirects to OAuth relier.
+
+### What happens when an unverified client requests `/certificate/sign`?
+
+If the sessionToken is unverified,
+the request will succeed.
+However, the verification status is stored
+in the certificate as `fxa-tokenVerified`,
+alongside other `fxa-` properties.
+The Sync server can subsequently reject assertions
+where `fxa-tokenVerified` is `false`.
+
+### What happens when an unverified client requests `/account/keys`?
+
+If the keyFetchToken is unverified,
+the request will fail with errno 102.
+
+### How does this affect users on legacy clients?
+
+They will receive the confirmation email,
+but they will not see the sign-in confirmation screen.
+Subsequent operations may fail with token errors
+until the confirm their email.
+
+**Shane, I wrote the above
+based on hazy recollection of what you said
+while I was only half-concentrating.
+Can you do fix this bit up?**
+
+## Work breakdown
+
+### fxa-content-server
+
+- [ ] Add strings to `strings.js`
+  to get localization done quicker.
+  This task is the highest immediate priority.
+- [ ] Implement "confirm your email" screen.
+- [ ] Implement confirmation landing screen.
+- [ ] Add handling for `verified` and `challenge` fields
+  in `/account/login` response.
+- [ ] Add handling for 102 errors
+  from `/certificate/sign`
+  and `/account/keys` endpoints.
+
+### fxa-auth-server
+
+- [ ] Add method to `mailer.js`
+  for sending confirmation email.
+- [ ] Modify `/account/login`
+  to create keys unverified,
+  initiate verification email
+  and set `challenge` on response.
+- [ ] Modify `/account/create`, `/account/openid/login` and `/password/change/start`
+  to always create their keys in the verified state.
+- [ ] Create a `verifiedSessionToken`
+  auth strategy for Hapi
+  that fails with 102 (unverified user) error
+  if sessionToken is not verified.
+- [ ] Create a `verifiedKeyFetchToken`
+  auth strategy for Hapi
+  that fails with 102 (unverified user) error
+  if keyFetchToken is not verified.
+- [ ] In `/certificate/sign`,
+  encode the sessionToken verification state
+  on the certificate as `fxa-tokenVerified`.
+- [ ] In `/account/keys`,
+  use new `verifiedKeyFetchToken` auth strategy.
+- [ ] Add a `/token/verify` endpoint for token verification.
+  The `tokenVerificationId` should be sent in the payload
+  as `code`, for consistency with `/recovery_email/verify_code`.
+- [ ] Modify the `/recovery_email/verify_code` endpoint
+  to also verify tokens.
+  This is in case an unverified user signs in,
+  we need their comfirmation to verify everything.
+- [ ] Modify the `/recovery_email/resend_status` endpoint
+  to handle both account verification and token verification emails
+  (and know the difference).
+
```

Comment:
I outlined above [how I imagine legacy devices](https://github.com/mozilla/fxa/pull/143/files#r59873372) will work, they'll receive a `verified: false` field in the response to `/account/login`, they'll then redirect to their confirm screen where they'll poll `/recovery_email/status` with the sessionToken returned from `/account/login`. `/recovery_email/status` will see the sessionToken is unverified and return `verified: false`. That will enable both FxOS and Firefox desktop to continue polling the auth server for until the user re-verifies their email.